core/modules/widgets/vars.js

```diff
@@ -0,0 +1,79 @@
+/*\
+title: $:/core/modules/widgets/vars.js
+type: application/javascript
+module-type: widget
+
+This widget allows the user to set multiple variables in one go.
+It thereby reduces code complexity that would arise when setting
+many variables using "<$set>".
+
+```
+\define helloworld() Hello world!
+<$vars greeting="Hi" me={{!!title}} sentence=<<helloworld>>>
+  <<greeting>>! I am <<me>> and I say: <<sentence>>
+</$vars>
+```
+
+Original discussion that led to the creation of this widget:
+https://github.com/Jermolene/TiddlyWiki5/issues/1610
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+var Widget = require("$:/core/modules/widgets/widget.js").widget;
+
+var VarsWidget = function(parseTreeNode,options) {
+	// Call the constructor
+	Widget.call(this);
+	// Initialise	
+	this.initialise(parseTreeNode,options);
+};
+
+/*
+Inherit from the base widget class
+*/
+VarsWidget.prototype = Object.create(Widget.prototype);
+
+/*
+Render this widget into the DOM
+*/
+VarsWidget.prototype.render = function(parent,nextSibling) {
+	this.parentDomNode = parent;
+	this.computeAttributes();
+	this.execute();
+	this.renderChildren(parent,nextSibling);
+};
+
+/*
+Compute the internal state of the widget
+*/
+VarsWidget.prototype.execute = function() {
+	// Parse variables
+	$tw.utils.each(this.attributes,function(val,key) {
+		if(key.charAt(0) !== "$") {
```

Comment:
Nope. I just thought it would be good to reserve it to avoid clashes with variable names. I mean, this is our only chance to add attributes that are not interpreted as variables :D